backend/utils/chat_notification.py

```diff
@@ -0,0 +1,218 @@
+event = Events()
+
+class Notification:
+    """
+    A class that handles the Novu instance message notification 
+    trigger for Channels, DM and when user is tagged.
+    This class is called after the message instance or object has been created
+    """  
+
+    async def check_message_tag(
+        self, message_obj, payload, 
+        sender_id
+        ):
+        """
+        A method that creates a notification trigger for tagged users
+        Args:
+            message_object-> Dict
+            payload-> Dict: message data that should be passed to novu
+            sender_id -> str
+        Raise:
+            HTTP_422- when Novu couldn't send notification to tagged users
+        """
+        tagged_users = []
+        # user_msg_tag = message_obj["richUiData"]['entityMap']
+        user_msg_tag = message_obj.get("entityMap", " ")
+        if user_msg_tag:
+            for i in range(len(user_msg_tag)):
+                member_id = user_msg_tag[str(i)]['data']['mention']['member_id']
+                if member_id == sender_id:
+                    del i[member_id]
+                tagged_users.append(member_id)
+            dm_trigger_create = await event.trigger(
+                'you-have-been-mentioned',
+                {
+                    "to": tagged_users,
+                    "payload": payload
+                })
+            if dm_trigger_create['statusCode'] !=201:
+                raise HTTPException(
+                    status_code=422, 
+                    detail="Novu couldn't send notifications to tagged users"
+                    )
+            return dm_trigger_create
+
+
+
+    async def dm_subscriber_create(self, room_obj):
+        """
+        A function that subscribes a DM user to Novu
+        Args:
+            (i) room_obj -> Dict
+
+        Raise:
+            HTTP_422- when novu failed to create DM notification
+            HTTP_400- when room object argument passed is invalid
+        """
+        if not room_obj:
+            return ValueError("Room object data is invalid")
+        if room_obj['room_type'] != 'DM':
+            # i would love to break out of the function if type not a DM
+            raise HTTPException(
+                status_code=400, 
+                detail="This function is only for DM"
+                )
+        try:
+            room_members = room_obj['room_members']
+        except:
+            raise HTTPException(
+                status_code=400, 
+                detail="room members can't be empty"
+            )
+        for member_id, value in room_members:
+            create_subcriber = await subscriber.identify(member_id)
+            if create_subcriber['statusCode'] !=201:
+                raise HTTPException(
+                status_code=422, 
+                detail="Novu couldn't create a Novu subscription for Novu"
+                )
+        return create_subcriber
+        
+        
+
+    async def channel_subcriber_create(self, member_id):
```

Comment:
how do you know what channel or room to subscribe the user/member to?

Reply:
The purpose of this task is to subscribe users on the Novu DB. since every room has members, I get the room ID from the message object, gets the members which definitely would have become a Novu subscriber either via DM or joining a channel and then send notification to this users. @mikengr

Reply:
This sounds like an assumption... `which definitely would have`??

Reply:
Not an assumption, maybe I picked the wrong choice of word but what I'm insinuating is, you'll become a Novu subscriber while trying to join a room. All that's needed is to send message to members in a room since each room as an ID and respective members @mikengr